lib/spack/docs/packaging_guide.rst

```diff
@@ -1806,6 +1892,46 @@ See the :ref:`concretization-preferences` section for more details.
 
 .. _install-method:
 
+------------------
+Inconsistent Specs
+------------------
+
+Suppose a user needs to install package C, which depends on packages A
+and B.  Package A builds a library with a Python2 extension, and
+pacakge B builds a library with a Python3 extension.  Packages A and B
+cannot be loaded together in the same Python runtime:
+
+.. code-block:: none
+
+    class A(Package):
+        variant('python', default=True, 'enable python bindings')
+        depends_on('python@2.7', when='+python')
+        def patch(self):
+            # do whatever is necessary to enable/disable python
+            # bindings according to variant
+
+    class B(Package):
+        variant('python', default=True, 'enable python bindings')
+        depends_on('python@3.2:', when='+python')
+        def patch(self):
+            # do whatever is necessary to enable/disable python
+            # bindings according to variant
```

Comment:
The patches don't seem to add anything import to this code snippet. Maybe you meant install?